src/infrastructure/jira/jira-service.ts

```diff
@@ -173,6 +171,100 @@ class JiraService {
 			}
 		}
 	};
+
+	deleteDesignUrlInIssueProperties = async (
+		issueIdOrKey: string,
+		design: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		await Promise.all([
+			await this.deleteAttachedDesignUrlInIssuePropertiesIfPresent(
+				issueIdOrKey,
+				design,
+				connectInstallation,
+			),
+			await this.deleteFromAttachedDesignUrlV2IssueProperties(
+				issueIdOrKey,
+				design,
+				connectInstallation,
+			),
+		]);
+	};
+
+	/**
+	 * @internal
+	 * Only visible for testing. Please use {@link deleteDesignUrlInIssueProperties}
+	 */
+	deleteAttachedDesignUrlInIssuePropertiesIfPresent = async (
+		issueIdOrKey: string,
+		design: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		try {
+			const response = await jiraClient.getIssueProperty(
+				issueIdOrKey,
+				propertyKeys.ATTACHED_DESIGN_URL,
+				connectInstallation,
+			);
+
+			if (response.value === design.url) {
+				await jiraClient.deleteIssueProperty(
+					issueIdOrKey,
+					propertyKeys.ATTACHED_DESIGN_URL,
+					connectInstallation,
+				);
+			}
+		} catch (error) {
+			if (error instanceof JiraClientNotFoundError) {
+				return; // Swallow not found errors
+			}
+			throw error;
+		}
+	};
+
+	/**
+	 * @internal
+	 * Only visible for testing. Please use {@link saveDesignUrlInIssueProperties}
+	 */
+	deleteFromAttachedDesignUrlV2IssueProperties = async (
+		issueIdOrKey: string,
+		{ url, displayName }: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		try {
+			const response = await jiraClient.getIssueProperty(
+				issueIdOrKey,
+				propertyKeys.ATTACHED_DESIGN_URL_V2,
+				connectInstallation,
+			);
+
+			const storedAttachedDesignUrlIssuePropertyValues =
+				response.value as AttachedDesignUrlV2IssuePropertyValue[];
+
+			const issuePropertyValueToRemove: AttachedDesignUrlV2IssuePropertyValue =
```

Comment:
nit: Is this `const` necessary? Looks like we only read the `url` key in the below filter `issuePropertyValueToRemove.url`, we could probably just use `url` directly and optionally rename it in the args de-structuring, i.e.

```
{ url: urlToRemove, displayName }: AtlassianDesign,
```

```
const newAttachedDesignUrlIssuePropertyValue =
		storedAttachedDesignUrlIssuePropertyValues.filter(
			({ url }) => url !== urlToRemove,
		);
```